streaming/src/main/scala/org/apache/spark/streaming/dstream/DStream.scala

```diff
@@ -490,6 +494,19 @@ abstract class DStream[T: ClassTag] (
     logDebug("Cleared checkpoint data")
   }
 
+  private[streaming] def readyToShutdown(): Unit = {
+    _readyToShutdown = true
+    dependencies.foreach(_.readyToShutdown())
+    logDebug("Ready to shutdown")
```

Comment:
Will it cause many log output since it's called in a recursive way? Since we have one in `DStreamGraph`, can we remove this?